x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/calculations/utils.ts

```diff
@@ -134,3 +134,35 @@ export function dateBasedOperationToExpression(
     },
   ];
 }
+
+/**
+ * Creates an expression ast for a date based operation (cumulative sum, derivative, moving average, counter rate)
+ */
+export function optionallHistogramBasedOperationToExpression(
+  layer: IndexPatternLayer,
+  columnId: string,
+  functionName: string,
+  additionalArgs: Record<string, unknown[]> = {}
+): ExpressionFunctionAST[] {
+  const currentColumn = (layer.columns[columnId] as unknown) as ReferenceBasedIndexPatternColumn;
+  const buckets = layer.columnOrder.filter((colId) => layer.columns[colId].isBucketed);
+  const nonHistogramColumns = buckets.filter(
+    (colId) =>
+      layer.columns[colId].operationType !== 'date_histogram' &&
+      layer.columns[colId].operationType !== 'range'
```

Comment:
It's kind of unusual to exclude `range` operations from this, it's not something we do for the other group-by logic. I think this should be consistent with the rest so I'm inclined to remove the `range` filter.

Reply:
I added this so it's possible to use overall calculations on number histograms. I think it makes sense because number histograms are often a distribution of some sort and it makes sense to treat it as a single data series in respect to overall calculation. This is an important use case for o11y.